lib/gslab_model/test/test.ExampleData.R

```diff
@@ -0,0 +1,100 @@
+source("ExampleData.R")
+
+x <- rnorm(1000)
+y <- rnorm(1000)
+z <- rnorm(1000)
+w <- rnorm(1500)
+rhs <- list(x = x, y = y)
+strvar <- c("f", "m", "m", "f", "m")
+group1 <- sort(sample(1:10, 1000, replace = TRUE))
+group2 <- sort(sample(1:10, 800, replace = TRUE))
+group3 <- sample(1:10, 1000, replace = TRUE)
+arrayVar1 <- array(rnorm(3000), c(1000, 3))
+arrayVar2 <- array(rnorm(6000), c(1000, 2, 3))
+
+test.ExampleData.initialize <- function() {
+    a <- ExampleData(x, y, const = list(mu = 0, sigma = 1))
+    checkEquals(c(a$varnames, a$nobs, a$nvars, a$const$mu), c(c("x", "y", "obsindex"), 1000, 3, 0))
+    b <- ExampleData(rhs, z, varnames = c("rhs1", "rhs2", "rhs3"))
+    checkEquals(c(b$varnames, b$nobs, b$nvars), c(c("rhs1", "rhs2", "rhs3", "obsindex"), 1000, 4))
+    c <- ExampleData(strvar, stringsAsFactors = FALSE, varnames = "gender")
+    checkEquals(c(c$varnames, c$nobs, c$nvars, class(c$var$gender)), c(c("gender", "obsindex"), 5, 2, "character"))
+    checkException(ExampleData(x, y, varnames = c("few")), silent = TRUE)
+}
+
+test.ExampleData.setGroup <- function() {
+    a <- ExampleData(x, y)
+    a$setGroup(group1)
+    checkEquals(a$ngroup, 10)
+    a <- ExampleData(x, y, group1)
+    a$setGroup(a$var$group1)
+    checkEquals(a$var$group1, a$groupvar)
+    checkException(a$setGroup(group2), silent = TRUE)
+    checkException(a$setGroup(group3), silent = TRUE)
+}
+
+test.ExampleData.addData <- function() {
```

Comment:
Currently if we add the same variable twice, it just change the name from, say, `x` to `x.1` (the default name for `data.frame` when the variable names are duplicated.) Do you think we need to add a check before to see if the added variable is identical in all values to any of the variable already in the dataset?